pkg/kubelet/pod_container_deletor.go

```diff
@@ -80,20 +81,27 @@ func getContainersToDeleteInPod(filterContainerId string, podStatus *kubecontain
 
 	// Find the exited containers whose name matches the name of the container with id being filterContainerId
 	var candidates containerStatusbyCreatedList
+	var infraCandidates containerStatusbyCreatedList
 	for _, containerStatus := range podStatus.ContainerStatuses {
 		if containerStatus.State != kubecontainer.ContainerStateExited {
 			continue
 		}
 		if matchedContainer == nil || matchedContainer.Name == containerStatus.Name {
-			candidates = append(candidates, containerStatus)
+			if containerStatus.Name == leaky.PodInfraContainerName {
+				infraCandidates = append(infraCandidates, containerStatus)
+			} else {
+				candidates = append(candidates, containerStatus)
+			}
 		}
 	}
 
 	if len(candidates) <= containersToKeep {
-		return containerStatusbyCreatedList{}
+		// Always destroy infra containers to ensure networking gets torn down
```

Comment:
kubelet still relies on one dead container as a checkpoint. I don't think we should change that. 

Overall, I'd prefer not making change to this file (`pod_container_deletor.go`). The podContainerDeletor simply does a more aggressive cleanup, and the regular container GC will kick in later. 

It's also a plus to not pollute more files with leaky.PodInfraContainerName